relay/pushservice/pushservice.py

```diff
@@ -0,0 +1,66 @@
+import json
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import messaging
+
+from relay.events import Event
+from relay.api.schemas import UserCurrencyNetworkEventSchema
+from .client_token_db import ClientTokenDB
+
+
+class FirebaseRawPushService:
+    """Sends push notifications to firebase. Sending is done based on raw client tokens"""
+
+    def __init__(self) -> None:
+        self._app = None
+
+    def initialize(self, path_to_keyfile: str) -> None:
+        """
+        Initializes the push service
+        Args:
+            path_to_keyfile: Path to json keyfile with firebase credentials
+        """
+        cred = credentials.Certificate(path_to_keyfile)
+        self.app = firebase_admin.initialize_app(cred)
+
+    def send_event(self, client_token, event: Event):
+        message = self._build_event_message(client_token, event)
+        messaging.send(message, app=self._app)
```

Comment:
We have to make sure that this is compatible with gevent. We don't use any monkeypatching at the moment AFAICT.

Reply:
If the library doesn't use a custom C extension, we should be pretty safe with monkey patching.

Reply:
So monkey patch all at the beginning of the relay.py file?